core/src/test/java/org/apache/iceberg/util/TestParallelIterable.java

```diff
@@ -133,6 +140,47 @@ public CloseableIterator<Integer> iterator() {
         .untilAsserted(() -> assertThat(queue).as("Queue is not empty after cleaning").isEmpty());
   }
 
+  @Test
+  public void limitQueueSize() throws IOException, IllegalAccessException, NoSuchFieldException {
```

Comment:
Did we check the effects of this on any of our JMH benchmarks? I'm assuming there is little change there right, but it would make sense to show it in the PR.

Reply:
Do you have any particular benchmark in mind?
I though that maybe `ManifestReadBenchmark` or `PlanningBenchmark` would be useful, but they don't seem to be. (my verification method: i planted an exception in the parallel iterator task code, but benchmarks still run)

Reply:
https://github.com/apache/iceberg/pull/10691#issuecomment-2239458430